packages/next/lib/typescript/writeConfigurationDefaults.ts

```diff
@@ -57,7 +59,7 @@ function getDesiredCompilerOptions(
       // All of these values work:
       parsedValues: [
         ts.ModuleResolutionKind.NodeJs,
-        ts.ModuleResolutionKind.Node12,
+        ts.ModuleResolutionKind.Node16,
```

Comment:
See https://github.com/microsoft/TypeScript/pull/48879

Reply:
There is no `Node12` anymore. 👀

https://github.com/microsoft/TypeScript/pull/48879/files#diff-dd6d044fa72b2a5303fe2eb63efcf13ffbc5acc4420e5a4b1224ac4df7667e57R2896